cmd/daemon.go

```diff
@@ -49,11 +51,20 @@ var startDaemonCmd = &cobra.Command{
 			return fmt.Errorf("daemon must be run as root")
 		}
 
-		var err error
+		ctx, cancel := context.WithCancel(cmd.Context())
+		defer cancel()
 
 		log.Info().Str("version", rootCmd.Version).Msg("starting daemon")
 
-		server, err := server.NewServer(cmd.Context(), &server.ServeOpts{
+		// Initialize metrics reporting if ASR flag is enabled
+		if viper.GetBool("metrics.asr") {
+			cedanaURL := viper.GetString("connection.url")
+			if err := metrics.PollAndPublishMetrics(ctx, cedanaURL); err != nil {
```

Comment:
I'd also move this initialization inside `server.Launch` itself. We initialize Otel stuff too in there, for example.